src/utp_stream.cpp

```diff
@@ -1687,9 +1691,12 @@ bool utp_socket_impl::send_pkt(int const flags)
 #endif
 		}
 	}
+	// If this is an ack then note that it still needs to be sent
+	else if (p->size == p->header_size && (flags & pkt_ack))
+		defer_ack();
 
 	// Any queued up deferred ack is now redundant
-	if (m_deferred_ack)
+	if (m_deferred_ack && !m_stalled)
```

Comment:
this if-block should probably just be moved into the `if (!m_stalled)` block above

Reply:
> this if-block should probably just be moved into the `if (!m_stalled)` block above

Done